fenicsprecice/adapter_core.py

```diff
@@ -274,3 +354,186 @@ def get_forces_as_point_sources(fixed_boundary, function_space, coupling_mesh_ve
 
     return x_forces.values(), y_forces.values()  # don't return dictionary, but list of PointSources
 
+
+def determine_shared_vertices(comm, rank, function_space, owned_gids, unowned_gids):
```

Comment:
Function name is not ideal. Maybe
```suggestion
def determine_shared_vertices_ownership_map(comm, rank, function_space, owned_gids, unowned_gids):
```
And this `map` is the dictionary mapping `global_ids` of shared `vertices` to the owning `rank`?